### ISSUE TITLE
[BUG] Winget unable to identify current version, and delayed updates

### ISSUE THREAD

**Issue by Sollaze** [radj307/volume-control#104]
### Version

6.3.0 and below

### Windows Version

Windows 10 Pro 22H2

### Actual Behaviour

As you can see in the screenshot below, winget is unable to identify the current version of the application and is also behind on the release versions (at least for some time).
![image](https://github.com/radj307/volume-control/assets/146455740/b969d54f-c0f3-4986-b03a-817c2302c13c)

### Expected Behaviour

Ideally, winget should be able to determine the current application version but it would also be great if it could detect (or deliver) updates as fast as the app's built-in update checker.

**Comment by radj307** [comment]
Not sure what kind of black magic winget is using.

Getting a lot of errors & unpredictable behaviour with https://github.com/vedantmgoyal2009/winget-releaser and only 1 workflow has succeeded. The pull request https://github.com/microsoft/winget-pkgs/pull/122992 is stalled due to a hash mismatch error, and yet winget can actually install v6.3.0 correctly.
![image](https://github.com/radj307/volume-control/assets/1927798/638016ae-3388-49b4-94b0-6ee397c97f0a)

I'm considering making a manifest & submitting a PR manually without using the 3rd party workflow.

https://github.dev/radj307/volume-control/blob/main/.github/workflows/winget.yml
https://github.com/radj307/volume-control/actions/runs/6635807157/job/18027310670
https://github.com/vedantmgoyal2009/winget-releaser/blob/main/main.ts

**Comment by radj307** [comment]
Related: https://github.com/microsoft/winget-cli/issues/1255

**Comment by radj307** [comment]
I have identified the problem. Winget gets the current version number from the program's Uninstall key in the Registry, by checking for a property named `DisplayVersion`.

Volume Control's registry key when installed for all users is here on my system:  
`Computer\HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall\{33DFCEE8-022C-4C66-A366-79A7415320F2}_is1`  
It contains these properties:  
![image](https://github.com/radj307/volume-control/assets/1927798/4219d91c-54ab-41ee-a8a2-f290f4661441)

The Inno setup script should be modified to set the `DisplayVersion` property to the installed version number, which should fix the problem.

**Comment by radj307** [comment]
https://jrsoftware.org/ishelp/index.php?topic=setup_appversion

**Comment by radj307** [comment]
Implemented a fix in d0320561af555401c1af2ebf3087a8686d249554

Despite Inno Setup providing a few different parameters to set various values in the application uninstall key, AFAIK it does not provide a way to set the `DisplayVersion` value. It does, however, provide a way to set registry keys/values in general.  
Since Volume Control can be installed for all users or only the current user, its registry key can be in two different places:  
`HKLM\Software\Microsoft\Windows\CurrentVersion\Uninstall` (All Users)  
`HKCU\Software\Microsoft\Windows\CurrentVersion\Uninstall` (Current User)  

Adding `PrivilegesRequired=lowest` to `[Setup]` will install the application without admin permissions when installing for the current user. Now we can check `IsAdmin` to determine the installation scope, and conditionally install the registry value to the appropriate location:  
```iss
[Registry]
; All Users
Root: HKLM; Subkey: "Software\Microsoft\Windows\CurrentVersion\Uninstall\{{#AppID}_is1"; ValueType: string; ValueName: "DisplayVersion"; ValueData: "{#AppVersion}"; Flags: uninsdeletevalue; Check: IsAdmin
; Current User
Root: HKCU; Subkey: "Software\Microsoft\Windows\CurrentVersion\Uninstall\{{#AppID}_is1"; ValueType: string; ValueName: "DisplayVersion"; ValueData: "{#AppVersion}"; Flags: uninsdeletevalue; Check: not IsAdmin
```

**Comment by radj307** [comment]
Due to the way that winget is set up, it just isn't possible to make new versions available very quickly.  

Volume control's built-in update checker works by comparing the version number of the latest release on github to the current version. In order for new releases to make it to winget, a release has to be posted to github, which then triggers the workflow to create a pull request in the winget-pkgs repository. The pull request has to be reviewed by a community moderator, which can take some time.

**Comment by radj307** [comment]
The changes in d0320561af555401c1af2ebf3087a8686d249554 did not fix the problem. The `DisplayVersion` registry key is created successfully, but Inno Setup immediately overrides it because it creates the uninstall key *after* anything set in the `[Registry]` section.

Related: https://stackoverflow.com/questions/60289363/write-custom-string-value-into-software-microsoft-windows-currentversion-uninsta

**Comment by radj307** [comment]
Fixed in fb0992ae7b1c3fda9575c37b8ea8ef0cee2de634
